lib/zcap/index.js

```diff
@@ -60,3 +63,69 @@ export async function create({
 
   return {rootCapability, encoded: `z${encoded}`};
 }
+
+/**
+ * Delegates an Authorization Capability to a target delegate.
+ *
+ * @param {object} options - The options to use.
+ * @param {{encoded: string, bytes: Uint8Array}} options.zcapControllerKeySeed -
+ *   32 byte random seed value, multihash identified, and multibase-encoded.
+ * @param {string} options.controller - Capability controller DID.
+ * @param {Array<string>} [options.allow] - Optional list of allowed
+ *   actions or string specifying allowed delegated action. Default: ['read'] -
+ *   delegate all actions.
+ * @param {string} [options.url] - URL to invoke the Authorization Capability
+ *   against, aka the `invocationTarget`.
+ * @param {string} [options.ttl] - Optional time to live value for the
+ *   delegation.
+ * @param {string} options.delegatee - URL identifying the entity to
+ *   delegate to.
+ * @param {string} [options.capability] - Optional capability to delegate.
+ * @param {string} [options.invocationTarget] - Optional target, used for
+ *   attenuating (restricting) the parent zcap's target.
+ * @param {string} [options.method='key'] - DID method id.
+ *
+ * @returns {Promise<object>} - A promise that resolves to an object with
+ *   a root capability and its base58-encoded version.
+ */
+export async function delegate({
+  zcapControllerKeySeed, method, allow, url, controller, delegatee,
```

Comment:
My only regret with these APIs is that they are all being built presuming that a seed will be available instead of being built with a signer API -- that *may* be constructed from a seed. We went through a lot of pain to reach the point where our other APIs all took `signer` style APIs to ensure that we could seamlessly hook up HSMs and WebKMS systems (etc.) that provided signing APIs where secret key material was never exposed. If we reach a point where we want to do the same with these APIs we'll have to refactor all this again. IMO, we should have stuck with that pattern and used something like `CapabilityAgent` (or similar) to generate a signer API and passed that primitive around instead.

Reply:
@dlongley - always a valid concern. Don't worry though, I'm keeping the `signer` style in mind. It'll be addressed in the next few PRs.
(These current ones are just quick bite-sized iterations.)